aspnetcore/fundamentals/dependency-injection/samples/3.x/DependencyInjectionSample/Models/Operation.cs

```diff
@@ -4,21 +4,32 @@
 namespace DependencyInjectionSample.Models
 {
     #region snippet1
-    public class Operation : IOperationTransient, 
-        IOperationScoped, 
-        IOperationSingleton, 
+    public class Operation : IOperationTransient,
+        IOperationScoped,
+        IOperationSingleton,
         IOperationSingletonInstance
     {
-        public Operation() : this(Guid.NewGuid())
+        public Operation() : this(DateTime.Now.ToString("hh:mm:ss.fff"))
         {
+            DoWork();
         }
 
-        public Operation(Guid id)
+        public Operation(string id)
         {
             OperationId = id;
         }
 
-        public Guid OperationId { get; private set; }
```

Comment:
@dougbu @serpent5  I propose we move all these operations to the view. That's more realistic to show scope has the same value per request but transient doesn't. I could even add a middleware. 

Should I add a middleware? That seems more realistic.

Reply:
I'm not sure I understand the suggestion here. Can you explain it a bit more, please?

Reply:
The code behind calles each service twice. The `Operation` code has the same 3 services. I think it would be  better to show the output in the logger, not a view. Move the  `Operation` code  to middleware. It's just not realistic to call the same services twice each in a RP code behind. I was thinking it would be slightly better to move that to the RP view page, but I think moving it to middleware or a filter is better.